backend/user-service/psql.js

```diff
@@ -0,0 +1,39 @@
+require('dotenv').config();
+const { Pool } = require('pg');
+
+const pool = new Pool({
+    host     : process.env.PSQL_HOSTNAME,
+    database : process.env.PSQL_database,
+    user     : process.env.PSQL_USERNAME,
+    password : process.env.PSQL_PASSWORD,
+    port     : process.env.PSQL_PORT    
+});
+
+const createUserTableQueryIfNotExist = `
```

Comment:
When hosting in supabase, I want all the tables organized. Hence, I would like to request your help to create a schema called `user` to store the `users` table.

Also, the `users` table does not have enough details. Every user needs to have a name, if they are students they must have a major, course (in the future), and more importantly, I need a user_id primary key as it is more elegant to query by user_id instead of email.  

Furthermore, to ensure security, I think it is good to hash the user password on frontend side even before we send the API request to backend to prevent password being sent in raw. To have this feature, we therefore need to store a salt for hashing the password so that we know how to restore it.

Reply:
Resolved